zyngine/zynthian_layer.py

```diff
@@ -535,27 +535,31 @@ def restore_zs3(self, i):
 		zs3 = self.zs3_list[i]
 
 		if zs3:
-			#Load bank list and set bank
-			self.load_bank_list()
-			self.set_bank_by_name(zs3['bank_name'])
-			self.wait_stop_loading()
-
-			#Load preset list and set preset
-			self.load_preset_list()
-			self.set_preset_by_name(zs3['preset_name'])
-			self.wait_stop_loading()
-
-			#Refresh controller config
-			if self.refresh_flag:
-				self.refresh_flag=False
-				self.refresh_controllers()
+			# For non-LV2 engines, bank and preset can affect what controllers do.
+			# In case of LV2, just restoring the controllers ought to be enough, which is nice
+			# since it saves the 0.3 second delay between setting a preset and updating controllers.
+			if not self.engine.nickname.startswith('JV'):
+				#Load bank list and set bank
+				self.load_bank_list()
+				self.set_bank_by_name(zs3['bank_name'])
+				self.wait_stop_loading()
+
+				#Load preset list and set preset
+				self.load_preset_list()
+				self.set_preset_by_name(zs3['preset_name'])
+				self.wait_stop_loading()
+
+				#Refresh controller config
+				if self.refresh_flag:
+					self.refresh_flag=False
+					self.refresh_controllers()
+				sleep(0.3)
```

Comment:
Please, keep setting the bank and preset. When dealing with synths, most of times you use a preset, and although you make some adjustments, you want to know you are using this preset when doing a ZS3 change. Only the sleep can be avoided.

Reply:
Well, probably we could also avoid the load_banks, and if the bank doesn't change, the set_bank & load_preset too. If the preset is also the same, then we could avoid the load preset too, but we must add the checks ...

Reply:
I've brought them back, but skip over them if no bank or preset has ever been selected in that snapshot (which reflects my usage at least :-).

Reply:
This seems OK. I will test it ASAP.